### ISSUE TITLE
Deepfake Image & Video Detection using CNN + Vision Transformer (ViT) Comparison

### ISSUE THREAD

**Issue by uv05709** [abhisheks008/DL-Simplified#1042]
### Deep Learning Simplified Repository (Proposing new issue)

🔴 **Project Title** : Deepfake Image & Video Detection using CNN + Vision Transformer (ViT) Comparison

🔴 **Aim** :
Deepfakes — AI-generated synthetic media that replace or manipulate a person's face in images/videos — are one of the most critical threats in digital trust today. The aim of this project is to build and compare multiple deep learning models to detect whether an image or video frame is **real or deepfake**, using a publicly available dataset. The project will benchmark at least 3 model architectures and identify which performs best on this binary classification task.

🔴 **Dataset** :
- Primary: [140k Real and Fake Faces — Kaggle](https://www.kaggle.com/datasets/xhlulu/140k-real-and-fake-faces)
- Supplementary: [FaceForensics++ subset](https://github.com/ondyari/FaceForensics) (for video frame extraction)
- Both are publicly available and free to use.

🔴 **Approach** :

This project will follow a structured multi-model comparison approach:

**Step 1 — Data Preprocessing**
- Load and normalize images (224×224)
- Apply augmentation: horizontal flip, rotation, brightness jitter
- Split: 80% train / 10% val / 10% test
- Class balance check (real vs fake distribution)

**Step 2 — Model 1: Custom CNN Baseline**
```python
# Lightweight 5-layer CNN as baseline
model = Sequential([
    Conv2D(32, (3,3), activation='relu', input_shape=(224,224,3)),
    MaxPooling2D(),
    Conv2D(64, (3,3), activation='relu'),
    MaxPooling2D(),
    Conv2D(128, (3,3), activation='relu'),
    Flatten(),
    Dense(256, activation='relu'),
    Dropout(0.5),
    Dense(1, activation='sigmoid')  # binary: real/fake
])
```

**Step 3 — Model 2: Transfer Learning with EfficientNetB0**
- Fine-tune pretrained EfficientNetB0 (ImageNet weights)
- Unfreeze last 20 layers for domain adaptation
- Binary classification head

**Step 4 — Model 3: Vision Transformer (ViT-B/16)**
- Use `vit-keras` or HuggingFace `ViTForImageClassification`
- Patch size 16×16, 12 transformer encoder layers
- Compare attention maps vs CNN feature maps

**Step 5 — Evaluation & Comparison**

| Metric | CNN | EfficientNetB0 | ViT-B/16 |
|--------|-----|----------------|----------|
| Accuracy | — | — | — |
| AUC-ROC | — | — | — |
| F1 Score | — | — | — |
| Inference time (ms) | — | — | — |

**Step 6 — Explainability with Grad-CAM**
- Apply Grad-CAM on best model to visualize which facial regions expose the deepfake
- Overlay heatmaps on sample predictions

**Step 7 — README with visualizations**
- Training curves, confusion matrices, sample predictions with heatmaps
- Conclusions on which model generalizes best

---

### 📍 Follow the Guidelines to Contribute in the Project :
- Separate folder: `Deepfake Detection/`
- Sub-folders: `Images/`, `Dataset/`, `Model/`
- `requirements.txt` included
- `README.md` filled using the standard template with visualizations

---

🔴🟡 **Points to Note :**
- 1 Issue == 1 PR
- Issue Title and PR Title must match (include issue number)
- Follow Contributing Guidelines & Code of Conduct

---

INTERMEDIATE | GSSoC | Status: Up for Grabs

**Comment by abhisheks008** [comment]
Hi @uv05709, similar kind of project is already present in this repository. Here it is: https://github.com/abhisheks008/DL-Simplified/tree/main/Deepfake%20Images%20Detector

- What makes this project standout from the existing one?
- Suggest atleast 4 models instead of 3.
- Can you share a detailed overview of the models along with the suitability of the models with the dataset structure you are choosing.

**Comment by Varshith-Mandala** [comment]
Hi @abhisheks008 , I’m highly interested in working on this issue under GSSoC’26.

This project stands out because it goes beyond a standard binary deepfake classifier and focuses on a comparative analysis between CNN-based architectures and Vision Transformers (ViTs), which is highly relevant to current real-world deepfake detection research. Instead of evaluating a single pipeline, the project can provide deeper insight into how spatial feature extraction, attention mechanisms, generalization ability, and robustness differ across architectures when handling manipulated media.

What makes this project more impactful than many existing implementations is the possibility of:

* Comparing both convolution-based and transformer-based approaches under the same preprocessing and evaluation pipeline
* Studying model robustness against compression artifacts, low-resolution frames, lighting variation, and facial manipulations
* Evaluating computational tradeoffs such as inference speed, parameter count, and training efficiency
* Analyzing whether transformer-based global attention provides better forgery detection compared to local CNN feature extraction
* Building a reproducible benchmarking framework instead of only a prediction model

I would like to expand the comparison from 3 models to at least 5 strong architectures to make the evaluation more meaningful and research-oriented.

Proposed Models:

1. CNN-Based Models

   * ResNet50
   * EfficientNet-B0/B3
   * XceptionNet

2. Transformer-Based Models

   * Vision Transformer (ViT)
   * Swin Transformer

Detailed Model Overview & Suitability:

1. ResNet50

* Strong baseline CNN architecture with residual connections
* Effective for extracting hierarchical facial texture features
* Good starting benchmark for deepfake classification
* Suitable for medium-scale datasets and faster experimentation

2. XceptionNet

* Widely used in deepfake research literature
* Excellent at learning subtle compression and manipulation artifacts
* Depthwise separable convolutions improve efficiency while preserving strong feature extraction
* Highly suitable for face-focused datasets like FaceForensics++

3. EfficientNet

* Provides strong accuracy-to-parameter efficiency ratio
* Scales effectively while reducing computational overhead
* Suitable when balancing performance and training cost is important

4. Vision Transformer (ViT)

* Uses self-attention instead of convolution
* Better at learning global relationships across image patches
* Potentially stronger for identifying inconsistencies distributed across the face or frame
* Requires larger datasets or transfer learning for stable performance

5. Swin Transformer

* Hierarchical transformer architecture with shifted windows
* More computationally efficient than standard ViT
* Better scalability for higher-resolution image inputs and video frame analysis
* Suitable for structured visual patterns and localized manipulation artifacts

Dataset Structure Proposal:
I propose using a structured dataset pipeline with:

* Real vs Fake binary classification
* Balanced class distribution
* Frame extraction for video-based datasets
* Separate train/validation/test splits to avoid data leakage
* Optional augmentation pipeline for robustness

Potential datasets:

* FaceForensics++
* Celeb-DF
* DeepFake Detection Challenge (DFDC)

Pipeline Plan:

1. Data Collection & Preprocessing

   * Face extraction and alignment
   * Frame sampling from videos
   * Image normalization and augmentation

2. Model Training

   * Unified training setup for fair comparison
   * Transfer learning using pretrained weights
   * Hyperparameter tuning and regularization

3. Evaluation

   * Accuracy
   * Precision
   * Recall
   * F1-score
   * ROC-AUC
   * Confusion matrix

4. Comparative Analysis

   * CNN vs Transformer performance
   * Training stability
   * Generalization capability
   * Computational efficiency
   * Robustness against low-quality manipulations

5. Possible Extensions

   * Grad-CAM visual explainability
   * Real-time inference benchmarking
   * Ensemble approach combining CNN + Transformer outputs

I’d be excited to contribute a well-structured implementation with proper experimentation, benchmarking, and documentation to make this project more research-focused and practically valuable.
